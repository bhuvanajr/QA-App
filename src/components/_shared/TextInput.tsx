import { ErrorMessage } from '@hookform/error-message';
import React from 'react';
import { useFormContext } from 'react-hook-form';
import { ErrorMessageDisplay } from './ErrorMessageDisplay';

type Props = {
  rules?: Object;
  name: string;
  className?: string;
} & React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>;

export const TextInput = (props: Props) => {
  const {
    register,
    formState: { errors },
  } = useFormContext();

  return (
    <>
      <input {...register(props.name, props.rules)} {...props} className={props.className} />
      {props.rules && (
        <ErrorMessage
          errors={errors}
          name={props.name}
          render={({ message }): React.ReactNode => <ErrorMessageDisplay id={props.name} message={message} />}
        />
      )}
    </>
  );
};
