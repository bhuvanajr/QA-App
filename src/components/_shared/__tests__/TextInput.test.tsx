import { render, screen } from '@testing-library/react';
import React from 'react';
import { TextInput } from '../TextInput';

jest.mock('../ErrorMessageDisplay.tsx', () => ({
  ErrorMessageDisplay: () => <div>ErrorMessageDisplay</div>,
}));

jest.mock('react-hook-form', () => ({
  ...jest.requireActual('react-hook-form'),
  useFormContext: () => ({
    register: () => jest.fn(),
    formState: () => jest.fn(),
  }),
}));

jest.mock('@hookform/error-message', () => ({
  ErrorMessage: () => <div>ErrorMessage</div>,
}));

test('should render TextInput correctly', () => {
  const { container } = render(<TextInput name="testInput" />);
  expect(container).toMatchSnapshot();
});

test('should render TextInput with ErrorMessage correctly when rules is passed', () => {
  const { container } = render(<TextInput name="testInput" rules={{ required: true }} />);
  expect(container).toMatchSnapshot();
  const errorMessage = screen.getByText(/ErrorMessage/i);
  expect(errorMessage).toBeInTheDocument();
});
