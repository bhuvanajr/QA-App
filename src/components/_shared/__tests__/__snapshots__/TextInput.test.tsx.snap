// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TextInput correctly 1`] = `
<div>
  <input
    mock="[object Object]"
    name="testInput"
  />
</div>
`;

exports[`should render TextInput with ErrorMessage correctly when rules is passed 1`] = `
<div>
  <input
    mock="[object Object]"
    name="testInput"
    rules="[object Object]"
  />
  <div>
    ErrorMessage
  </div>
</div>
`;
